src/transformers/models/t5/modeling_t5.py

```diff
@@ -934,9 +934,9 @@ def forward(
                     encoder_extended_attention_mask = encoder_extended_attention_mask.to(hidden_states.device)
                 if encoder_decoder_position_bias is not None:
                     encoder_decoder_position_bias = encoder_decoder_position_bias.to(hidden_states.device)
-                if head_mask is not None:
+                if not (isinstance(head_mask, list) and head_mask[0] is None):
```

Comment:
After calling `get_head_mask`, `head_mask` cannot be `None` but only be a list of `None`'s